test/object-safe-get/index.js

```diff
@@ -57,29 +69,74 @@ test('returns undefined for non-existing properties, using array arg', function(
   t.end();
 });
 
+test('returns defaultValue (third parameter) for non-existing properties, using array arg', function(t) {
+  t.plan(8);
+  var obj = {a: {aa: {aaa: 2}}, b: 4, c: null, d: 0};
+  t.ok(compare(get(obj, ['b', 'bb'], 888), 888));
+  t.ok(compare(get(obj, ['a', 'bb'], 888), 888));
+  t.ok(compare(get(obj, ['b', 'bb', 'bbb'], 888), 888));
+  t.ok(compare(get(obj.b, ['bb', 'bbb'], 888), 888));
+  t.ok(compare(get(obj, ['c', 'cc'], 888), 888));
+  t.ok(compare(get(obj, ['d', 'dd', 'ddd'], 888), 888));
+  var arr = ['b', 'bb', 'bbb'];
+  t.ok(compare(get(obj, arr, 888), 888));
+  t.ok(compare(arr, ['b', 'bb', 'bbb'])); // array arg preserved
+
+  t.end();
+});
+
 test('returns undefined for falsey property names using dot notation', function(t) {
-  t.plan(4);
+  t.plan(5);
   var obj = {a: {aa: {aaa: 2}}, b: {}};
-  t.ok(compare(get(obj, 'a, ""', undefined)));
-  t.ok(compare(get(obj, 'a, aa, aaa, ""', undefined)));
-  t.ok(compare(get(obj, 'b, ""', undefined)));
-  t.ok(compare(get(obj, 'b, ""'), undefined));
+  t.ok(compare(get(obj, 'a.'), undefined));
+  t.ok(compare(get(obj, 'a.aa.aaa.'), undefined));
+  t.ok(compare(get(obj, 'b.'), undefined));
+  t.ok(compare(get(obj, 'b..b'), undefined));
+  t.ok(compare(get(obj, 'b...b'), undefined));
```

Comment:
I added those tests just to tick the "pathological cases with dots", but in fact, it returns `undefined` only because those objects do not have `""` (empty string) props. When a dot is followed by nothing or by another dot, the lib tries to read `""` empty string prop of the (sub)object. Should we codify this behavior here?

See the last test case I added

Reply:
I think this is correct. For every prop (where a prop is any element formed by splitting the argument by dot) we do obj[prop] where prop is a string. We can only create string properties from a string argument so we assume the prop between `..` is an empty string.

This is identical to what happens when an array argument is supplied, the only difference is an array argument can supply non-string props. The effect is the same, an empty string is just another falsey value, and (matching lodash, I think) when any of the provided props is falsey, the result is always `undefined`.